tests/components/bryant_evolution/test_climate.py

```diff
@@ -0,0 +1,399 @@
+"""Test the BryantEvolutionClient type."""
+
+import asyncio
+from collections.abc import Callable
+from dataclasses import dataclass
+import logging
+from unittest.mock import patch
+
+from evolutionhttp import BryantEvolutionLocalClient
+import pytest
+
+from homeassistant.components.bryant_evolution.const import CONF_SYSTEM_ZONE, DOMAIN
+from homeassistant.components.climate import (
+    ATTR_FAN_MODE,
+    ATTR_HVAC_ACTION,
+    SERVICE_SET_FAN_MODE,
+    SERVICE_SET_HVAC_MODE,
+    SERVICE_SET_TEMPERATURE,
+    HVACAction,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.const import ATTR_ENTITY_ID, CONF_FILENAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+from homeassistant.util.unit_system import US_CUSTOMARY_SYSTEM
+
+from tests.common import MockConfigEntry
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def _wait_for_cond(predicate: Callable[[], bool]) -> None:
+    """Wait for `predicate` to return True."""
+    while True:
+        if predicate():
+            break
+        await asyncio.sleep(0.1)
+
+
+class _FakeEvolutionClient(BryantEvolutionLocalClient):
```

Comment:
I have tried the mock objects in the past before, but in my experience they always turned to be a PITA. I now always tend to recommend to make use of mocks and that really helps you decouple the tests from the library. After migrating `twitch` to mocks, I am kinda convinced everything can be mocked (they use a lot of async expressions I never knew). So can you maybe please elaborate on why you went this route before I overlook some design choices?